software/firmware/bootloader.py

```diff
@@ -78,29 +82,43 @@ def get_class_for_name(script_class_name) -> type:
             )
             return None
 
-    @classmethod
-    def load_script_classes(cls, scripts) -> "dict(str, type)":
-        classes = {}
-        for i, script in enumerate(scripts):
-            with PrintMemoryUse(script):
-                clazz = cls.get_class_for_name(script)
-            if clazz:
-                classes[script] = clazz
-            cls.show_progress(i / len(scripts))
-        return classes
-
     @classmethod
     def _is_europi_script(cls, c):
         return issubclass(c, EuroPiScript)
 
-    @staticmethod
-    def _build_scripts_mapping(classes):
-        return OrderedDict(
-            [(cls.display_name(), cls) for cls in classes if BootloaderMenu._is_europi_script(cls)]
-        )
+    @classmethod
+    def show_error(self, title, message, duration=1):
+        """Show a brief error message on-screen saying we can't launch the requested script
+
+        @param title    The title of the error
+        @param message  The body of the error message
+        @param duration The number of seconds the message should show. If negative the message is shown forever
+        """
+        oled.fill(0)
+        oled.centre_text(f"--{title}--\n{message}")
+        oled.show()
+
+        if duration > 0:
+            time.sleep(duration)
+        else:
+            while True:
+                time.sleep(1)
```

Comment:
[discussion] I'm not sure that I see the need for a loop here. The message will stay on the screen as long as nothing else draws over it, so why bother with an infinite loop? I think it might be simpler to have this method format and display the message, and leave the timing up to the caller.

Reply:
This was done basically as an equivalent to a kernel panic or BSOD function; the point is to permanently prevent any other execution.  The idea being that this _could_ be triggered from inside another script if there's an error to functionally terminate the script in a safe way that would also prevent any changes to the output voltages.

It's probably overkill, but that was the motivation behind it implementing it this way.

Reply:
OK, I tried this out and I like the results, there is likely some value to giving the user an error that they can report.

However, I commented out the entire `else` block to remove the loop and the behavior didn't change. So I still don't think the need the loop and sleep.

Reply:
As long as there's nothing else writing to the OLED the sleep shouldn't be needed.

I'm wondering if it's worth having the top-level exception-handler set all outputs to zero?